tests/migrations/test_operations.py

```diff
@@ -2074,6 +2074,41 @@ def test_remove_field(self):
         self.assertEqual(definition[1], [])
         self.assertEqual(definition[2], {"model_name": "Pony", "name": "pink"})
 
+    def test_remove_field_non_nullable(self):
+        """
+        Tests the RemoveField operation with explicit default
+        """
+        project_state = self.set_up_test_model("test_rmflnn")
+        operation = migrations.RemoveField("Pony", "weight")
+        new_state = project_state.clone()
+        Pony = project_state.apps.get_model("test_rmflnn.Pony")
+        Pony.objects.create(pink=1, weight=1337)
+        operation.state_forwards("test_rmflnn", new_state)
+
+        with connection.schema_editor() as editor:
+            operation.database_forwards("test_rmflnn", editor, project_state, new_state)
+
+        # MySQL populates NOT NULL cols with implicit defaults instead of
+        # raising an error.
```

Comment:
It's crazy to think that MySQL [still does that](https://dbfiddle.uk/5odHM7dM) and doesn't have a `sql_mode` to entirely disable it.

It seems that [STRICT_ALL_TABLES](https://dev.mysql.com/doc/refman/8.4/en/sql-mode.html#sqlmode_strict_all_tables) doesn't apply here and `STRICT_TRANS_TABLE` is used

> If a value is missing, MySQL inserts the implicit default value for the column data type.

Here are [the docs](https://dev.mysql.com/doc/refman/8.4/en/data-type-defaults.html#data-type-defaults-implicit) about it.